crates/astria-sequencer/src/main.rs

```diff
@@ -1,17 +1,26 @@
 use astria_sequencer::{
-    Config,
+    config,
     Sequencer,
 };
 use tracing::info;
 
+// Following the BSD convention for failing to read config
+// See here: https://freedesktop.org/software/systemd/man/systemd.exec.html#Process%20Exit%20Codes
+const EX_CONFIG: i32 = 78;
+
 #[tokio::main]
 async fn main() {
-    let config = Config::get();
-    telemetry::init(std::io::stdout, config.log.as_deref().unwrap_or("info"))
-        .expect("failed to initialize telemetry");
+    let config = match config::get() {
+        Ok(cfg) => cfg,
+        Err(e) => {
+            eprintln!("failed to read configuration: {e:?}");
+            std::process::exit(EX_CONFIG);
```

Comment:
I have had a read about `process::exit` and think we should avoid it, because it immediately stops the process without running any destructors. That's not a huge deal in this case, but seems bad practice in general (future PRs might add some code before the config result gets evaluated, not paying attention the exit).

It's probably better to change `async fn main() -> std::process::ExitCode`

Have a look at the conductor refactor here and what its main.rs does: https://github.com/astriaorg/astria/pull/392